### PR TITLE
Wiggle sort in C

### LINKED ISSUE #5197: Wiggle sort in C
## 🚀 Feature

To reorder an unsorted array such that arr[0]<arr[1]>arr[2]<arr[3]...

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### PR DESCRIPTION
**Pull request by aish2002**
Fixes #5197 

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Wiggle sort in C

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#5197

### CODE REVIEW

**Comment by aish2002** [comment]
@ankitaggarwal23 , @HarshCasper  kindly review the PR

File: C/sort/Wiggle_sort.c [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    for (int m = 0; m < k; m++)
+        ar[l++] = c[m];
+}
+
+//to sort the array using merge sort
+void merge_sort(int *ar, int l, int r)
+{
+    if (l < r)
+    {
+        int mid = (l + r) / 2;
```
Comment (atarax665) [inline C/sort/Wiggle_sort.c]: Use ```mid = l + (r - l)  /  2``` to avoid overflow.

File: C/sort/Wiggle_sort.c [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    for (int k = 0; k < n; k += 2)
+        // left is decremented
+        arr[k] = res[left--];
+
+    //Elements in the right sub-array are added to the odd indices of vector x
+    for (int k = 1; k < n; k += 2)
+        // right is decremented
+        arr[k] = res[right--];
+
+    printf("[");
```
Comment (atarax665) [inline C/sort/Wiggle_sort.c]: No need of enclosing the output sorted array in the bracket, you can print them normally.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by aish2002** [comment]
I have made the necessary changes. @atarax665 , @ankitaggarwal23  kindly re-review the PR

**Review by ankitaggarwal23 — COMMENTED** [review]
Don't push more than one code file in a single PR.
Closing this PR as of now. Open the new PR if you want this to be reviewed and add the proper Files.

**Pull request closed without merging**